packages/steps/src/__tests__/BuildConfig-test.ts

```diff
@@ -33,7 +33,7 @@ describe(readAndValidateBuildConfigAsync, () => {
     const config = await readAndValidateBuildConfigAsync(
       path.join(__dirname, './fixtures/build.yml'),
       {
-        externalFunctionIds: [],
+        externalFunctionIds: ['eas/save-cache'],
```

Comment:
I can see `eas/*-cache` functions being added as external functions in at least a couple of tests. Why wouldn't they be added as EAS Functions (in `steps/easFunctions.ts`)?

Reply:
`readAndValidateBuildConfigAsync` doesn't know about eas functions, these are passed in `runCustomBuildAsync`(https://github.com/expo/eas-build/blob/afe570422d2c45a7478786e1aeb885a5014d90cf/packages/build-tools/src/builders/custom.ts#L39 and https://github.com/expo/eas-build/blob/afe570422d2c45a7478786e1aeb885a5014d90cf/packages/steps/src/BuildConfigParser.ts#L67).

Reply:
Hm, maybe let me rephrase my curiosity. Why do we only need to add `eas/save-cache` here and not all other EAS functions? Like, why this function is special and needs to be added here?

Reply:
Oh, that's because the file that we parse (`./fixtures/build.yml`) only contains steps using either `run` expressions or `eas/save-cache` function. Only functions that are used in template are checked for existence, and `run` is a special case (https://github.com/expo/eas-build/blob/afe570422d2c45a7478786e1aeb885a5014d90cf/packages/steps/src/BuildConfig.ts#L435)

Reply:
Aaaaah, it's a test file. Yea, sorry, didn't see that.